libs/ui-lib/lib/ocm/components/clusterConfiguration/ExternalPlatformDropdown.tsx

```diff
@@ -15,62 +16,112 @@ import {
   VSPHERE_CONFIG_LINK,
   getFieldId,
 } from '../../../common';
+import { ExternalPlatformIntegrationStateMap, FeatureIdPlatform } from './types';
 
 const INPUT_NAME = 'platform';
 const fieldId = getFieldId(INPUT_NAME, 'input');
 export type ExternalPlatformType = Extract<PlatformType, 'none' | 'nutanix' | 'oci' | 'vsphere'>;
 
 type ExternalPlatformDropdownProps = {
   showOciOption: boolean;
-  disabledOciTooltipContent: React.ReactNode;
-  isOciDisabled: boolean;
   onChange: (selectedPlatform: ExternalPlatformType) => void;
-  dropdownIsDisabled: boolean;
+  cpuArchitecture?: string;
+  clusterExists?: boolean;
+  clusterPlatform?: PlatformType;
+  externalPlatformIntegrationStateMap: ExternalPlatformIntegrationStateMap;
 };
 
 export type ExternalPlatformInfo = {
   label: string;
   href: string;
   tooltip: string;
+  featureId: string;
 };
 
 export const externalPlatformTypes: Record<ExternalPlatformType, ExternalPlatformInfo> = {
   none: {
     label: 'No platform integration',
     href: '',
     tooltip: '',
+    featureId: '',
   },
   nutanix: {
     label: 'Nutanix',
     href: NUTANIX_CONFIG_LINK,
     tooltip: '',
+    featureId: 'NUTANIX_INTEGRATION',
   },
   oci: {
     label: 'Oracle  (Requires a custom manifest)',
     href: '',
     tooltip:
       "To integrate with an external partner (for example, Oracle Cloud), you'll need to provide a custom manifest.",
+    featureId: 'EXTERNAL_PLATFORM_OCI',
   },
   vsphere: {
     label: 'vSphere',
     href: VSPHERE_CONFIG_LINK,
     tooltip: '',
+    featureId: 'VSPHERE_INTEGRATION',
   },
 };
 
+export const areAllExternalPlatformIntegrationDisabled = (
+  externalPlatformIntegrationStateMap: ExternalPlatformIntegrationStateMap,
+): boolean => {
+  // Check if isDisabled is true for all options
+  return Object.values(externalPlatformIntegrationStateMap).every(
+    (info) => info.isDisabled === true,
+  );
+};
+
 export const ExternalPlatformDropdown = ({
   showOciOption,
-  disabledOciTooltipContent,
-  isOciDisabled,
   onChange,
-  dropdownIsDisabled,
+  cpuArchitecture,
+  externalPlatformIntegrationStateMap,
+  clusterPlatform,
 }: ExternalPlatformDropdownProps) => {
-  const [field, { value }, { setValue }] = useField<string>(INPUT_NAME);
+  const [field, , { setValue }] = useField<string>(INPUT_NAME);
   const [isOpen, setOpen] = React.useState(false);
+
+  const defaultValue = React.useMemo(() => {
+    if (clusterPlatform !== undefined) {
+      const platform = clusterPlatform === 'baremetal' ? 'none' : clusterPlatform;
+      return platform;
+    } else {
+      return 'none';
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [clusterPlatform]);
+  const [current, setCurrent] = React.useState<string>(defaultValue);
```

Comment:
Why do we need another field to hold the value?
What's the difference to  the fields's inner value? What if they get out of sync? eg. in the new `useEffect`, we're only setting `current` but not value.